app/utils/index.ts

```diff
@@ -17,3 +24,32 @@ export const parseFirebaseError = (error: string) => {
 export const getKeyFromLocalizationString = (localizationString: string) => {
   return localizationString.split('.').join('-');
 };
+
+/*
+ Parse string for example: Hello [child]!
+ and retern next srting array: [Hello, child, !]
+*/
+export const parseTextWithNickname = (
+  text: string,
+): IParseTextWithNicknameResult => {
+  const result = [];
+
+  const startNicknameTypeIndex = _.indexOf(text, NICKNAME_INSERT_SYMBOL_START);
+  const endNicknameTypeIndex = _.indexOf(text, NICKNAME_INSERT_SYMBOL_END);
+  const nicknameType: NicknameType =
+    text.substring(startNicknameTypeIndex + 1, endNicknameTypeIndex) === 'Child'
+      ? NicknameType.Child
+      : NicknameType.Parent;
+
+  if (startNicknameTypeIndex) {
+    result.push(text.substring(0, startNicknameTypeIndex));
+  }
+  result.push(nicknameType);
+  if (endNicknameTypeIndex !== text.length - 1) {
+    result.push(text.substring(endNicknameTypeIndex + 1));
+  }
+  return {
+    nicknameType: nicknameType,
```

Comment:
is it boolean value?
we need name it properly, better put here the type from `types` not `boolean` value

Reply:
has no logical meaning, you have made a mistake
```
export interface IParseTextWithNicknameResult {
  nicknameType: NicknameType;
  textArray: string[];
}
```